Yoast/wordpress-seo
# Fixes an error when the note_when_disabled attribute is missing on a feature toggle.

## Context


*

## Summary


This PR can be summarized in the following changelog entry:

* Fixes an error in the Premium integration tests when a `note_when_disabled` attribute was missing on a feature toggle.

## Relevant technical choices:

* 

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* Checkout `trunk` of the WordPress SEO Premium repository.
* Require this feature branch using Composer:
  ```
  composer require yoast/wordpress-seo dev-fix-note_when_disabled-attribute-missing@dev
  ```
* Setup the WordPress develop environment for running the integration tests in Premium.
   * [You can follow these steps to set one up](https://developer.yoast.com/development/environment/setup-plugin-integration-tests).
* Run `composer integration-test` in WordPress SEO Premium. All tests should pass.  
* visit the Redirects page on a subsite in a multisite environment
  * without this patch, you should see a notice such as
```
NOTICE: wp-content/plugins/wordpress-seo/admin/class-yoast-form.php:878 - Undefined index: note_when_disabled
do_action('seo_page_wpseo_redirects'), WP_Hook->do_action, WP_Hook->apply_filters, WPSEO_Redirect_Page->display, WPSEO_Redirect_Page_Presenter->display, include('/plugins/wordpress-seo-premium/classes/redirect/views/redirects.php'), WPSEO_Redirect_Tab_Presenter->display, include('/plugins/wordpress-seo-premium/classes/redirect/views/redirects-tab-settings.php'), Yoast_Form->toggle_switch
```
  * with this patch, you should not see any notice like that

#### Relevant test scenarios
  * See impact check below.  


### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

* you should use the current Premium `trunk` or a RC containing this PR: https://github.com/Yoast/wordpress-seo/pull/18729
* visit the Redirects page on a subsite in a multisite environment
  * without this patch, you should see a notice such as
```
NOTICE: wp-content/plugins/wordpress-seo/admin/class-yoast-form.php:878 - Undefined index: note_when_disabled
do_action('seo_page_wpseo_redirects'), WP_Hook->do_action, WP_Hook->apply_filters, WPSEO_Redirect_Page->display, WPSEO_Redirect_Page_Presenter->display, include('/plugins/wordpress-seo-premium/classes/redirect/views/redirects.php'), WPSEO_Redirect_Tab_Presenter->display, include('/plugins/wordpress-seo-premium/classes/redirect/views/redirects-tab-settings.php'), Yoast_Form->toggle_switch
```
  * with this patch, you should not see any notice like that
* Smoke test the feature toggles on single sites and multisites (see the impact check below).

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

* The feature toggles on single sites and multisites. Please check whether the feature toggles in the site admin (_Yoast SEO_ > _Settings_ > _Features_) work as expected and do not throw any PHP warnings or errors.
  * Specific things to check:
     * Disabling a feature network wide on a multisite environment. It should still show a disabled note on sub sites saying that the feature has been disabled network wide.
     * Disabling the inclusive language feature by setting the site's language to a language other than English.
       * In single site environments and multisite environments. 

## UI changes


## Other environments


## Documentation


## Quality assurance


Fixes #
